core/conn/spj_init/src/main/java/com/traf/mgmt/JarFileMgmt.java

```diff
@@ -0,0 +1,435 @@
+			file.delete();
+			LOG.info("Remove " + fileName + " successfully!");
+			return;
+		} else {
+			throw new SQLException("No such file[" + fileName + "]");
+		}
+	}
+
+	/**
+	 * Remove files via regular formulation
+	 * 
+	 * @param pattern:
+	 *            to be deleted
+	 * @param names
+	 *            : file names to be deleted
+	 * @throws SQLException
+	 */
+	public static void rmRex(String pattern, String[] names) throws SQLException {
+		checkFileName(pattern);
+		Connection conn = getConn();
+		LOG.info("Try to remove files[" + pattern + "]");
+		String userPath = getCodeFilePath(conn);
+		close(conn);
+		File[] files = getFiles(pattern, new File(userPath));
+		StringBuilder sb = new StringBuilder();
+		sb.append("<rmRex>");
+		sb.append(toXML(files, "rmList"));
+		sb.append("<message>");
+		boolean hasError = false;
+		for (File f : files) {
+			try {
+				f.delete();
+			} catch (Exception e) {
+				hasError = true;
+				LOG.error(e.getMessage(), e);
+				sb.append("<error fileName='" + f.getName() + "'>" + e.getMessage() + "</error>");
+			}
+		}
+		if (!hasError) {
+			sb.append("Remove the files successfully!");
+		}
+		sb.append("</message>");
+		sb.append("</rmRex>");
+		names[0] = sb.toString();
+		LOG.info("Done for removing files[" + pattern + "].");
+	}
+
+	public static void lsAll(String[] names) throws SQLException {
+		ls("*", names);
+	}
+
+	/**
+	 * list the Jars matching PATTERN
+	 * 
+	 * @param pattern:
+	 * @param names
+	 * @throws SQLException
+	 */
+	public static void ls(String pattern, String[] names) throws SQLException {
+		checkFileName(pattern);
+		Connection conn = getConn();
+		LOG.info("List files[" + pattern + "]");
+		String userPath = getCodeFilePath(conn);
+		close(conn);
+		File dir = new File(userPath);
+		if (!dir.exists() || !dir.isDirectory()) {
+			LOG.error("Directory [" + userPath + "] is not found!");
+			throw new SQLException("Directory [" + userPath + "] is not found!");
+		}
+		if (pattern == null) {
+			LOG.error("File pattern should not be empty!");
+			throw new SQLException("Pattern is empty!");
+		}
+		File[] files = getFiles(pattern, dir);
+		names[0] = toXML(files, "ls");
+	}
+
+	/**
+	 * upload a JAR file
+	 * 
+	 * @param fileData
+	 * @param fileName
+	 * @param appendFlag
+	 *            0: append; otherwise overwrite
+	 * @throws SQLException
+	 */
+	public static void put(String fileData, String fileName, int appendFlag) throws SQLException {
+		checkFileName(fileName);
+		try {
+			byte[] data = fileData.getBytes(CHARTSET);
+
+			Connection conn = getConn();
+			LOG.info("Put " + fileName + ", length: " + data.length + ", file string length:" + fileData.length());
+			String userPath = getCodeFilePath(conn);
+			close(conn);
+			String fname = userPath + fileName;
+			checkFile(fname, data.length);
+			FileOutputStream fos = null;
+			try {
+				fos = new FileOutputStream(fname, (appendFlag == 0));
+
+				fos.write(Snappy.uncompress(Arrays.copyOf(data, data.length)));
```

Comment:
IMHO it would be better to just forget about Snappy and send the data uncompressed. Requiring Snappy has caused us a lot of trouble on the server side. Requiring it on the client and server side (probably with compatible versions) is not worth it for this use case that's about convenience, not performance.